extras/log_file_client/lib/utils/log_list.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:flutter/services.dart';
+import 'package:html/parser.dart';
+import 'package:http/http.dart' as http;
+
+class Log {
+  Log(this.name, this.uri);
+  final String name;
+  final String uri;
+}
+
+abstract class LogListReader {
+  Future<List<Log>> fetchList();
+}
+
+class LogListReaderForTest implements LogListReader {
+  @override
+  Future<List<Log>> fetchList() async {
+    return [Log('test1', '/test1.test'), Log('test2', '/test2.test'), Log('test3','/test3.test')];
+  }
+}
+
+class LogListReaderForAppWeb implements LogListReader {
```

Comment:
This has a lot of shared code with the next class. Perhaps you could create a method in the abstract superclass that takes HTML and returns the `List<Log>`. In fact, you could modify the `fetchList` above to have a hard-coded HTML "file" that tests the new method.